src/Microsoft.AspNetCore.Mvc.Core/Internal/ApiBehaviorApplicationModelProvider.cs

```diff
@@ -91,6 +94,21 @@ public void OnProvidersExecuting(ApplicationModelProviderContext context)
             }
         }
 
+        internal static void AddGloballyConfiguredApiConventions(ControllerModel controllerModel)
+        {
+            if (controllerModel.Filters.OfType<ApiConventionAttribute>().Any())
+            {
+                // ApiControllerAttribute is already associated with controller. Do not look for conventions configured at assembly.
+                return;
+            }
+
+            var assembly = controllerModel.ControllerType.Assembly;
```

Comment:
Just a thought, it seems possible for users to define all their conventions in an assembly which might not even have controllers.

Reply:
There's a difference between this and the the actual convention though. The attribute is expected to live on the controller assembly, but the convention can live elsewhere e.g. DefaultApiConventions live in Mvc.